mne/io/eyelink/tests/test_eyelink.py

```diff
@@ -317,3 +316,20 @@ def test_annotations_without_offset(tmp_path):
     assert raw.annotations[1]["description"] == "SYNCTIME"
     assert_allclose(raw.annotations[-1]["onset"], onset1)
     assert_allclose(raw.annotations[1]["onset"], onset2 - 2 / raw.info["sfreq"])
+
+
+@requires_testing_data
+def test_no_datetime(tmp_path):
+    """Test reading a file with no datetime."""
+    out_file = tmp_path / "tmp_eyelink.asc"
+    with open(fname) as file:
+        lines = file.readlines()
+    # remove the timestamp from the datetime line
+    lines[1] = lines[1].split(":")[0]
```

Comment:
I think this won't quite do what you want as the colon will be gone:
```
>>> "DATE: etc".split(":")[0]
'DATE'
```
and hence `line.startswith("** DATE:")` conditional will never be hit. So instead:
```suggestion
    lines[1] = lines[1].split(":")[0] + ":"
```

Reply:
Yeah I agree that your suggestion makes the intent of the code more clear, thanks 🙏 

FWIW  the `_get_recording_datetime` method will return `None` even if that `line.startswith("** DATE:")` conditional is never hit, which should make the code robust to scenarios where that datetime line is just entirely missing from the file (i.e.  that's why my test was passing before this suggestion).

... But I guess in _that_ scenario, the user will never be warned (logging statement is never hit).